src/main/java/com/airbnb/kafka/KafkaStatsdMetricsReporter.java

```diff
@@ -23,119 +21,130 @@
     private String host;
     private int port;
     private String prefix;
+    private long pollingPeriodInSeconds;
     private MetricDimensionOptions metricDimensionOptions;
     private MetricPredicate metricPredicate;
-
-
-    @Override
-    public String getMBeanName() {
-        return "kafka:type="+getClass().getName();
-    }
+    private StatsDClient statsd;
+    private boolean isTagSupported;
+    private AbstractPollingReporter underlying = null;
 
     public boolean isRunning() {
-        return running;
+        return running.get();
     }
 
+    //try to make it compatible with kafka-statsd-metrics2
     @Override
     public synchronized void init(VerifiableProperties props) {
+        loadConfig(props);
+        if (enabled) {
+            log.info("Reporter is enabled and starting...");
+            startReporter(pollingPeriodInSeconds);
+        } else {
+            log.warn("Reporter is disabled");
+        }
+    }
+
+    private void loadConfig(VerifiableProperties props) {
         enabled = props.getBoolean("external.kafka.statsd.reporter.enabled", false);
-        int pollingIntervalSecs = props.getInt("kafka.metrics.polling.interval.secs", 10);
         host = props.getString("external.kafka.statsd.host", "localhost");
         port = props.getInt("external.kafka.statsd.port", 8125);
-        prefix = props.getString("external.kafka.statsd.metrics.prefix", null);
-        metricPredicate = MetricPredicate.ALL;
-        metricDimensionOptions = MetricDimensionOptions.fromProperties(props.props(), "external.kafka.statsd.dimension.enabled.");
-
-        String exclude_regex = props.getString("external.kafka.statsd.metrics.exclude_regex", DEFAULT_EXCLUDE_REGEX);
-        if (exclude_regex.length() != 0){
-            metricPredicate = new ExcludeMetricPredicate(exclude_regex);
+        prefix = props.getString("external.kafka.statsd.metrics.prefix", "");
+        pollingPeriodInSeconds = props.getInt("kafka.metrics.polling.interval.secs", 10);
+        metricDimensionOptions = MetricDimensionOptions.fromProperties(props.props()
+                , "external.kafka.statsd.dimension.enabled.");
+
+        String regexFilter = props.getString("external.kafka.statsd.metrics.exclude_regex", DEFAULT_REGEX_FILTER);
+        if (regexFilter != null && regexFilter.length() != 0) {
+            metricPredicate = new ExcludeMetricPredicate(regexFilter);
         }
 
-
-        underlying = new StatsDReporter(Metrics.defaultRegistry(), host, port, prefix, metricPredicate, metricDimensionOptions);
-
-        if (enabled) {
-            if (logger.isInfoEnabled())
-                logger.info("Kafka Statsd metrics reporter is enabled");
-
-            startReporter(pollingIntervalSecs);
-        } else {
-            if (logger.isInfoEnabled())
-                logger.info("Kafka Statsd metrics reporter is disabled");
-
-        }
+        this.isTagSupported = props.getBoolean("external.kafka.statsd.support.tag", true);
```

Comment:
I think `enabled` is more accurate than `supported`.
Can we rename the variable into `isTagEnabled` and the property to `external.kafka.statsd.tag.enabled`